### PR TITLE
[ADD] website: ability to restrict groups on website menu

### PR DESCRIPTION
**Pull request by JKE-be**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by JKE-be** [comment]
@robodoo r+

Thanks @Yenthe666

**Comment by Yenthe666** [comment]
🎉 co-authorship would've been nice but happy we get this in V13! Thanks.

**Comment by JKE-be** [comment]
Oups probably lost during a rebase...
Sorry

File: addons/website/models/website_menu.py [hunk 1]
```diff
@@ -26,6 +26,8 @@ def _default_sequence(self):
     child_id = fields.One2many('website.menu', 'parent_id', string='Child Menus')
     parent_path = fields.Char(index=True)
     is_visible = fields.Boolean(compute='_compute_visible', string='Is Visible')
+    group_ids = fields.Many2many('res.groups', string='Visible Groups',
```
Comment (sswapnesh) [inline addons/website/models/website_menu.py]: Rare case but still we can consider this as Improvement as we are adding new field.
Goog to add relation explicitly so we can prevent situation like this (https://github.com/odoo/odoo/pull/29376) in future.

**Comment by keshrath** [comment]
Maybe I'm doing something wrong, but the change didn't have the expected result for me.

No access rights (menu is visible - works as expected):

image

With access rights (menu is invisible, but shop is still accessible - works not as expected):

image

image

**Comment by JKE-be** [comment]
As the commit explains, it is to only restrict menu and not content for now.

**Comment by Yenthe666** [comment]
Indeed, it is only to remove the menuitems. If you need more advanced behaviour so that access can even be prevented you'll still need custom code. This is just so you can manipulate the menuitems from a functional point of view.

**Pull request closed without merging**